src/index.js

```diff
@@ -154,20 +135,43 @@ export default function normalize(obj, key=null) {
     return { entities, normalizations, result };
 }
 
-export function denormalize(key, builder, state, requiredFields) {
+/**
+* Denormalizes the entity that is associated with the key.
+*
+* @param {Object} key
+* @param {Builder} builder
+* @param {Map} state
+* @param {Function} validator Accepts three parameters (denormalized entity, entity key, key) and returns false if entity is invalid
+* @return {Array}|{Object}|{Void} denormalized entity or null if invalid or not found
+*/
+export function denormalize(key, builder, state, validator = null) {
     const entityKey = getEntityKey(builder);
     if (!state.entities[entityKey]) {
         return;
     }
     if (Array.isArray(key)) {
-        return key.map(id => {
+        let valid = true;
+        let entities = [];
+        key.every(id => {
```

Comment:
if we had this:

`[valid, invalid, valid]`

this loop would return True that the entities are valid.

if we had:

`[valid, valid, invalid]`

it would return `null`.

if we switch this to a normal for loop, we can bail as soon as we come across the first invalid entity.